python/pylibcugraph/pylibcugraph/bfs.pyx

```diff
@@ -0,0 +1,103 @@
+from libc.stdint cimport uintptr_t
+from libc.stdint cimport int64_t
+from libc.limits cimport INT_MAX
+
+from pylibcugraph.resource_handle cimport ResourceHandle
+from pylibcugraph._cugraph_c.algorithms cimport (
+    cugraph_bfs,
+    cugraph_paths_result_t,
+    cugraph_paths_result_get_vertices,
+    cugraph_paths_result_get_predecessors,
+    cugraph_paths_result_get_distances,
+    cugraph_paths_result_free,
+)
+from pylibcugraph._cugraph_c.array cimport (
+    cugraph_type_erased_device_array_view_t,
+    cugraph_type_erased_device_array_view_create,
+    cugraph_type_erased_device_array_view_free,
+)
+from pylibcugraph._cugraph_c.resource_handle cimport (
+    bool_t,
+    data_type_id_t,
+    cugraph_resource_handle_t,
+)
+from pylibcugraph._cugraph_c.error cimport (
+    cugraph_error_code_t,
+    cugraph_error_t,
+)
+from pylibcugraph.utils cimport (
+    assert_success,
+    copy_to_cupy_array,
+    assert_CAI_type,
+    get_c_type_from_numpy_type,
+)
+from pylibcugraph._cugraph_c.graph cimport (
+    cugraph_graph_t,
+)
+from pylibcugraph.graphs cimport (
+    _GPUGraph,
+)
+
+def bfs(ResourceHandle handle, _GPUGraph graph, 
+        sources, bool_t direction_optimizing, size_t depth_limit, 
+        bool_t compute_predecessors, bool_t do_expensive_check):
+    try:
```

Comment:
pylibcugraph.bfs will be available for users (with different experience than python cugraph users), so a docstring along with an example would be helpful for documentation purposes.

Reply:
sure, I will add that in